exporter/datadogexporter/traces_exporter.go

```diff
@@ -135,8 +138,11 @@ func (exp *traceExporter) exportHostMetrics(ctx context.Context, hosts map[strin
 			}
 			series = append(series, ms...)
 		}
-		ctx = clientutil.GetRequestContext(ctx, string(exp.cfg.API.Key))
-		_, _, err = exp.metricsAPI.SubmitMetrics(ctx, datadogV2.MetricPayload{Series: series})
+		_, err = exp.retrier.DoWithRetries(ctx, func(context.Context) error {
+			ctx = clientutil.GetRequestContext(ctx, string(exp.cfg.API.Key))
```

Comment:
```suggestion
			ctx := clientutil.GetRequestContext(ctx, string(exp.cfg.API.Key))
```
Probably not an issue, but never the less more correct? Try `ctx2` if the above fails because of some shadowing lint.

Reply:
Good catch!